cedar-policy/src/api.rs

```diff
@@ -245,111 +255,180 @@ impl Entities {
     }
 
     /// Create an `Entities` object with the given entities.
-    /// It will error if the entities cannot be read or if the entities hierarchy is cyclic
+    ///
+    /// `schema` represents a source of `Action` entities, which will be added
+    /// to the entities provided.
+    /// (If any `Action` entities are present in the provided entities, and a
+    /// `schema` is also provided, each `Action` entity in the provided entities
+    /// must exactly match its definition in the schema or an error is
+    /// returned.)
+    ///
+    /// If a `schema` is present, this function will also ensure that the
+    /// produced entities fully conform to the `schema` -- for instance, it will
+    /// error if attributes have the wrong types (e.g., string instead of
+    /// integer), or if required attributes are missing or superfluous
+    /// attributes are provided.
     pub fn from_entities(
         entities: impl IntoIterator<Item = Entity>,
+        schema: Option<&Schema>,
     ) -> Result<Self, entities::EntitiesError> {
         entities::Entities::from_entities(
             entities.into_iter().map(|e| e.0),
+            schema
+                .map(|s| cedar_policy_validator::CoreSchema::new(&s.0))
+                .as_ref(),
             entities::TCComputation::ComputeNow,
+            Extensions::all_available(),
         )
         .map(Entities)
     }
 
-    /// Add all of the [`Entity`]s in the collection to this [`Entities`] structure, re-computing the transitive closure
-    /// Re-computing the transitive closure can be expensive, so it is advised to not call this method in a loop
+    /// Add all of the [`Entity`]s in the collection to this [`Entities`]
+    /// structure, re-computing the transitive closure.
+    ///
+    /// If a `schema` is provided, this will inform the parsing: for instance, it
+    /// will allow `__entity` and `__extn` escapes to be implicit.
```

Comment:
Comment about json escapes is not relevant, since this is not a parsing function

Reply:
Fixed